### PR TITLE
Feature/search work

### PR DESCRIPTION
**Pull request by milkavladislav**
image
Create page with jobs and filters.

### CODE REVIEW

File: src/components/UI/SearchWorkForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import { FieldValues, useForm } from "react-hook-form";
+import LayoutElementWithTitle from "components/layoutElementWithTitle/LayoutElementWithTitle";
+import { useTranslation } from "react-i18next";
+import { IFreelancer } from "services/profileInfo/typesDef";
+import S from "components/UI/SearchWorkForm/styles";
+import propsDataCollection from "components/UI/SearchWorkForm/staticData";
+import { Typography } from "antd";
+import { useState } from "react";
+import { ISearchWorkFilters } from "./typesDef";
```
Comment (xx-TOOL-xx) [inline src/components/UI/SearchWorkForm/index.tsx]: absolute imports

File: src/components/UI/SearchWorkForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+interface ISearchWorkFormProps {
+  filters: ISearchWorkFilters;
+  setFilters: (filters: ISearchWorkFilters) => void;
+}
+
+const defaultCategoryId = 1;
+const defaultDurationType = "week";
```
Comment (xx-TOOL-xx) [inline src/components/UI/SearchWorkForm/index.tsx]: we have duration type in consts/enum. Use it

File: src/components/UI/SearchWorkForm/typesDef.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export interface ISearchWorkFilters {
+  take?: number;
+  skip?: number;
+  category?: number;
+  skills?: { id: number }[];
+  rate?: number;
+  duration?: number;
+  durationType?: string;
+  englishLevel?: string;
```
Comment (xx-TOOL-xx) [inline src/components/UI/SearchWorkForm/typesDef.ts]: we have union type TEnglishLevel and enum for duration type. Use them

File: src/components/UI/WorkCard/WorkCard.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  return (
+    <Card>
+      <JobTitle font_sz="1.5em">
+        {title}
+        <JobTitle font_sz="0.7em">
+          {createdAt ? dayjs(createdAt).fromNow() : t("NotFound.notFound")}
+        </JobTitle>
+      </JobTitle>
+      <JobDescription>
+        {jobDescription.split(" ").slice(0, 50).join(" ")}...
```
Comment (xx-TOOL-xx) [inline src/components/UI/WorkCard/WorkCard.tsx]: Just advice.
As I undestand u making this for words wrap, I believe it work for CSS

File: src/components/skillsLayout/style.ts [hunk 5]
```diff
@@ -1,8 +1,15 @@
...
+interface SDiv {
+  showSmall?: boolean;
+}
+
+export default styled("div")<SDiv>`
   display: grid;
-  grid-template-columns: repeat(4, 1fr);
+  grid-template-columns: repeat(
+    ${(props) => (props.showSmall ? "2" : "4")},
+    1fr
```
Comment (xx-TOOL-xx) [inline src/components/skillsLayout/style.ts]: according to what I see on screenshot, and in code above. U drilling props, creating new element in swith case, changing interfaces, and in the end all changes what u make it grid-template-column. Meanwhile all what u needed its writing here media-query for max-width. I dont insist on changes, but I totally disagree with approach like this.

File: src/components/layoutElementWithTitle/style.ts [hunk 6]
```diff
@@ -4,6 +4,7 @@ import { ILayoutElementWithTitleProps } from "components/layoutElementWithTitle/
 export const SWrapper = styled("div")`
   display: flex;
   align-items: flex-start;
+  flex-direction: column;
```
Comment (xx-TOOL-xx) [inline src/components/layoutElementWithTitle/style.ts]: Check please create job post page and freelancer edit profile page, and two another pages which uses this component. I believe that this one line will change user interface on all forms which maded before!!! Use media query with max width to implement this style only for cases when it small width, which is will work in this case

File: src/services/jobPosts/setJobPostsAPI.ts [hunk 7]
```diff
@@ -91,6 +93,18 @@ const jobPostsAPI = createApi({
         },
       }),
     }),
+    filterJobPosts: builder.query<
+      { data: IWorkCardProps[]; totalCount: number },
```
Comment (xx-TOOL-xx) [inline src/services/jobPosts/setJobPostsAPI.ts]: Move to interface. Watch on code below and make in same style, it's mach more easier to work with api query params type and api's return values type when their types declared

File: src/components/skillsLayout/style.ts (hunk 5 above)
Comment (milkavladislav) [reply src/components/skillsLayout/style.ts]: ohh, realy. I did not even think about it

File: src/components/layoutElementWithTitle/style.ts (hunk 6 above)
Comment (milkavladislav) [reply src/components/layoutElementWithTitle/style.ts]: Ohh, I forgot to change it

File: src/components/UI/SearchWorkForm/index.tsx [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    setTimeout(() => {
+      setIsReset(false);
+    }, 1000);
+  };
+
+  const { t } = useTranslation();
+
+  return (
+    <S.Form onSubmit={handleSubmit(onSubmit)}>
+      <Title level={3}>Filters</Title>
```
Comment (xx-TOOL-xx) [inline src/components/UI/SearchWorkForm/index.tsx]: move to localization

File: src/pages/Freelancer/SearchWork/Index.tsx [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+import { ISearchWorkFilters } from "components/UI/SearchWorkForm/typesDef";
+import { Element, scroller } from "react-scroll";
+
+const workPerPage = 10;
+const defaultPage = 1;
+
+function SearchWork() {
+  const { t } = useTranslation();
+  const [currentPage, setCurrentPage] = useState(defaultPage);
+  const [postsPerPage, setPostsPerPage] = useState(workPerPage);
```
Comment (xx-TOOL-xx) [inline src/pages/Freelancer/SearchWork/Index.tsx]: generic type, `useState<number>`

File: src/components/UI/SearchWorkForm/index.tsx [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+const defaultDurationType = DurationTypeEnum.WEEK;
+
+function SearchWorkForm({
+  filters,
+  setFilters,
+  setDefaultPage,
+}: ISearchWorkFormProps) {
+  const { Title } = Typography;
+
+  const [isReset, setIsReset] = useState(false);
```
Comment (xx-TOOL-xx) [inline src/components/UI/SearchWorkForm/index.tsx]: generic type

**Review by xx-TOOL-xx — APPROVED** [review]

File: src/components/UI/SearchWorkForm/index.tsx [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+      }),
+    });
+    setDefaultPage();
+  };
+
+  const onClear = () => {
+    setIsReset(true);
+    setFilters({ ...filters, skills: [] });
+    reset({
+      category: defaultCategoryId,
```
Comment (lyudad) [inline src/components/UI/SearchWorkForm/index.tsx]: nono no, what id therewill be no category with id1 in db?

File: src/components/UI/SearchWorkForm/index.tsx [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+      rate: "",
+      skills: [],
+      duration: "",
+      durationType: defaultDurationType,
+      englishLevel: undefined,
+    });
+    setTimeout(() => {
+      setIsReset(false);
+    }, 1000);
```
Comment (lyudad) [inline src/components/UI/SearchWorkForm/index.tsx]: delete magic number

File: src/components/UI/WorkCard/WorkCard.tsx [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+        }}
+      >
+        <JobTitle font_sz="1.5em">
+          {title}
+          <JobTitle font_sz="0.7em">
+            {createdAt ? dayjs(createdAt).fromNow() : t("NotFound.notFound")}
+          </JobTitle>
+        </JobTitle>
+        <JobDescription>
+          {jobDescription.split(" ").slice(0, wordPerCard).join(" ")}...
```
Comment (lyudad) [inline src/components/UI/WorkCard/WorkCard.tsx]: move this magic to utils, with function name that will describe what are you trying to do

**Pull request merged**